scripts/ci/run_perftest.sh

```diff
@@ -119,125 +161,155 @@ force_compaction() {
+  if [ -z $INFLUXDB2 ] || [ $INFLUXDB2 = true ]; then
+    load_opts="$load_opts -organization=$TEST_ORG -token=$TEST_TOKEN"
+  fi
+
+  # run ingest tests
+  $GOPATH/bin/bulk_load_influx $load_opts | \
+    jq ". += {branch: \"$INFLUXDB_VERSION\", commit: \"$TEST_COMMIT\", time: \"$datestring\", i_type: \"$DATA_I_TYPE\", use_case: \"$usecase\"}" > \
+      $working_dir/test-ingest-$usecase.json
+
+  # Cleanup
+  force_compaction
+  rm ${DATASET_DIR}/$data_fname
+done
 
-# Generate data
-scale_string="scalevar-$scale_var"
-scale_seed_string="$scale_string-seed-$seed"
-data_fname="influx-bulk-records-usecase-metaquery-$scale_seed_string.txt"
-$GOPATH/bin/bulk_data_gen --seed=$seed -use-case metaquery -scale-var $scale_var -timestamp-start $start -timestamp-end $end > ${DATASET_DIR}/$data_fname
+query_types() {
+  case $1 in
+    window-agg|group-agg|bare-agg)
+      echo min mean max first last count sum
+      ;;
+    metaquery)
+      echo field-keys tag-values
+      ;;
+    *)
+      echo "unknown use-case: $1"
+      exit 1
+      ;;
+  esac
+}
 
-# Generate flux queries for field-keys and tag-values
+# Generate queries to test.
 query_files=""
-for type in field-keys tag-values; do
-  for format in http flux-http; do
-    query_fname="query-metaquery-$scale_seed_string-$type-$format.txt"
-    $GOPATH/bin/bulk_query_gen -use-case metaquery -query-type $type -format influx-$format -timestamp-start $start -timestamp-end $end -queries $queries > ${DATASET_DIR}/$query_fname
-    query_files="$query_files $query_fname"
+for format in http flux-http; do
+  # Aggregate queries
+  for usecase in window-agg group-agg bare-agg metaquery; do
+    for type in $(query_types $usecase); do
+      query_fname="${format}_${usecase}_${type}"
+      $GOPATH/bin/bulk_query_gen \
+          -use-case=$usecase \
+          -query-type=$type \
+          -format=influx-$format \
+          -timestamp-start=$(start_time $usecase) \
+          -timestamp-end=$(end_time $usecase) \
+          -queries=$queries \
+          -scale-var=$scale_var > \
+        ${DATASET_DIR}/$query_fname
+      query_files="$query_files $query_fname"
+    done
   done
 done
 
-# Load the data
-load_opts="-urls=http://${NGINX_HOST}:8086 -do-abort-on-exist=false -do-db-create=true"
-if [ -z $INFLUXDB2 ] || [ $INFLUXDB2 = true ]; then
-  load_opts="$load_opts -organization=$TEST_ORG -token=$TEST_TOKEN"
-fi
-cat ${DATASET_DIR}/$data_fname | $GOPATH/bin/bulk_load_influx $load_opts > /dev/null
+org_flag() {
+  case $1 in
+    flux-http)
+      echo -organization=$TEST_ORG
+      ;;
+    http)
+      echo -use-compatibility=true
+      ;;
+    *)
+      echo echo "unknown query format: $1"
+      exit 1
+      ;;
+  esac
+}
 
-# compacting is skipped for now on metaquery tests, which have many shards
-# force_compaction
```

Comment:
Is it still intentional to skip forcing compaction here?

Reply:
This should be/is now compacting after both ingest runs. [This `find` call](https://github.com/influxdata/influxdb/pull/22068/files#diff-735f59fdcc40b53f4d07e10255120a973570da59f72757a690c62e4b30d87499R164-R166) works around the error that Will was seeing when trying to compact the metaquery data's shards.